packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -150,6 +150,9 @@ export class DeviceSession implements Disposable {
     ]);
     Logger.debug("App and preview ready, moving on...");
     this.eventDelegate.onStateChange(StartupMessage.AttachingDebugger);
+    if (this.debugSession) {
+      this.debugSession.dispose();
+    }
```

Comment:
would feel more correct for this to be in `startDebugger` method where we also re-assign debugSession variable immediately after disposing the previous one